### PR TITLE
feat(Message): replace referencedMessage with fetchReference

### PR DESCRIPTION
**Pull request by monbrey**
This replaces the cache-reliant `Message#referencedMessage` with a more reliable way to resolve the Message
This follows precedent set by #5480 

The method with throw an error if `Message#reference` is null, or if the channel to fetch the message from is not in the client's cache, as is likely to be the case when the message is a crosspost rather than a reply.

This seemed like the best way to handle it without getting into raw APIs calls and partial channels and all that mess.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- This PR changes the library's interface (methods or parameters added)
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: src/structures/Message.js [hunk 1]
```diff
@@ -406,15 +406,15 @@ class Message extends Base {
...
-   * @readonly
+   * Fetches the Message this crosspost/reply/pin-add references, if available to the client
+   * @returns {Promise<Message>}
    */
-  get referencedMessage() {
-    if (!this.reference) return null;
-    const referenceChannel = this.client.channels.resolve(this.reference.channelID);
-    if (!referenceChannel) return null;
-    return referenceChannel.messages.resolve(this.reference.messageID);
+  fetchReference() {
```
Comment (kyranet) [inline src/structures/Message.js:412]: This function needs to be marked as async (or use `return Promise.reject()`) so errors can be handled with `.catch()` handlers.
```suggestion
  async fetchReference() {
```

**Review by kyranet — CHANGES_REQUESTED** [review]

File: src/structures/Message.js (hunk 1 above)
Comment (monbrey) [reply src/structures/Message.js:412]: Should this also await on the last line? Linter will complain about an unused async otherwise.

Comment (ckohen) [reply src/structures/Message.js:412]: linter will complain about a useless await if you add it after a return. I think returen Promise.reject is a better option

Comment (kyranet) [reply src/structures/Message.js:412]: I'm pretty sure the rule checks for the `await` and `throw` keywords, as `async` changes their behaviour.

**Review by iCrawl — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**